### PR TITLE
Specter2: Spacing auto-annotate

### PR DESCRIPTION
**Pull request by lynnerang**
Allows designers to auto-annotate frame padding and the first gap measurement in frames with item spacing.  Few caveats to this:

- Only auto-annotates spacing if there are no spacing annotations yet.  This is because spacing is standalone/not tied to the original selection, so spacing annotations would be duplicated every time auto-annotate is run otherwise
- Doesn't annotate component instances, seeing as how many of them will be DS components with padding/gaps pre-determined
- Only annotates the first gap in item spacing, since they all match

Screen Shot 2022-08-08 at 11 49 43 AM

### CODE REVIEW

File: specter2/src/App.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ const getViewData = (
   });
 
   const listIds: Array<string> = listData.map(item => item.id);
-  let newSelections: Array<{ name: string, id: string, hasPadding?: boolean }> = [];
+  let newSelections: Array<{ name: string, id: string, isPadded?: boolean }> = [];
```
Comment (lynnerang) [inline specter2/src/App.ts:142]: naming is hard lol.  figured I'd go with 'isPadding' in this lil instance to open up 'hasPadding' for the function that checks for it 🤷‍♀️

File: specter2/src/App.ts [hunk 2]
```diff
@@ -769,10 +787,6 @@ export default class App {
             x: node.x,
             y: node.y,
           };
-
-          if (value !== getRoundedTokenVal(value, spacing.library === 'mercado')) {
-            figma.notify('Warning: Detected spacing does not match library spacing tokens.');
-          }
```
Comment (lynnerang) [inline specter2/src/App.ts:775]: This was taking precedence over the resulting toast for auto-annotation, and lasting forever when there were multiple cases where there wasn't an exact match, so gonna forgo it since we have the 'revert' icon next to the value in the annotation form.

File: specter2/src/App.ts [hunk 3]
```diff
@@ -610,32 +609,51 @@ export default class App {
...
     }
 
-    if (nodeLists.content.length || nodeLists.interaction.length) {
-      const numInteractions: number = nodeLists.interaction.length;
+    if (nodeLists.content.length || nodeLists.interaction.length || nodeLists.spacing.length) {
+      const numInteraction: number = nodeLists.interaction.length;
       const numContent: number = nodeLists.content.length;
-      figma.notify(`Success! ${numInteractions ? `${numInteractions} INTERACTION${numContent ? ' and ' : ''}` : ''}${numContent ? `${numContent} CONTENT ` : ''} annotations added`);
+      const numSpacing: number = nodeLists.spacing.length ? getData(frame, 'spacings')?.length : 0;
+      figma.notify(`Success! ${numSpacing ? `${numSpacing} SPACING${numContent || numInteraction ? ' and ' : ''}` : ''}${numInteraction ? `${numInteraction} INTERACTION${numContent ? ' and ' : ''}` : ''}${numContent ? `${numContent} CONTENT ` : ''} annotations added`);
```
Comment (gblakeman) [inline specter2/src/App.ts:656]: Might not much matter since this is just for a toast, but it's a bit hard to follow 3 ternaries in a row on a single line. Maybe set a few variables and feed the variable into the `notify` string?

**Review by gblakeman — APPROVED** [review]
🥳🎈

Noted a few small things. Thanks for the context comments!

File: specter2/src/App.ts (hunk 3 above)
Comment (lynnerang) [reply specter2/src/App.ts:656]: yeahhh figure it matters, it ended up getting extra gnarly when I added spacing in for this.  pushing an update in just a few that breaks it up into steps.
